libs/verify-android-environment.sh

```diff
@@ -51,4 +51,40 @@ if [ -z "${ANDROID_NDK_TOOLCHAIN_DIR}" ]; then
   exit 1
 fi
 
+# Determine the Java command to use to start the JVM.
+if [ -n "$JAVA_HOME" ] ; then
+    if [ -x "$JAVA_HOME/jre/sh/java" ] ; then
+        JAVACMD="$JAVA_HOME/jre/sh/java"
+    else
+        JAVACMD="$JAVA_HOME/bin/java"
+    fi
+    if [ ! -x "$JAVACMD" ] ; then
+        echo "ERROR: JAVA_HOME is set to an invalid directory: $JAVA_HOME
+
+Please set the JAVA_HOME variable in your environment to match the
+location of your Java installation."
+        exit 1
+    fi
+else
+    JAVACMD="java"
+    which java >/dev/null 2>&1 || die "ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
+
+Please set the JAVA_HOME variable in your environment to match the
+location of your Java installation."
+fi
+
+JAVA_MAJOR_VERSION=$($JAVACMD -version 2>&1 | sed -E -n 's/.* version "([^.-]*).*"/\1/p' | cut -d' ' -f1)
+if [[ "$JAVA_MAJOR_VERSION" -lt 8 ]] ; then
+    if [[ "$JAVA_MAJOR_VERSION" -eq 1 ]] ; then
+      version=$("$_java" -version 2>&1 | awk -F '"' '/version/ {print $2}')
```

Comment:
I think this needs to be `$JAVACMD` still. `$_java` isn't defined anywhere.

Surprised a bit that we don't have this set to error, but that can be addressed later.